packages/cli-kit/src/public/node/api/admin.ts

```diff
@@ -21,6 +23,34 @@ export async function adminRequest<T>(query: string, session: AdminSession, vari
   return graphqlRequest({query, api, url, token: session.token, variables})
 }
 
+/**
+ * Executes a GraphQL query against the Admin API. Uses typed documents.
+ *
+ * @param query - GraphQL query to execute.
+ * @param session - Shopify admin session including token and Store FQDN.
+ * @param variables - GraphQL variables to pass to the query.
+ * @param version - API version.
+ * @returns The response of the query of generic type <TResult>.
+ */
+export async function adminRequestDoc<TResult, TVariables extends Variables>(
+  query: TypedDocumentNode<TResult, TVariables>,
+  session: AdminSession,
+  variables?: TVariables,
+  version?: string,
+): Promise<TResult> {
+  let apiVersion = version
+  if (!version) {
+    apiVersion = await fetchLatestSupportedApiVersion(session)
+  }
+  const opts = {
+    url: adminUrl(session.storeFqdn, apiVersion),
+    api: 'Admin',
+    token: session.token,
+  }
+  const result = graphqlRequestDoc<TResult, TVariables>({...opts, query, variables})
+  return result
+}
+
```

Comment:
I've noticed the `fetchApiVersions` call had the `responseOptions: {handleErrors: false}`. It seems like that impacts how responses are handled and may control how they are [logged](https://github.com/Shopify/cli/blob/a2bb5689b8ad370774af9ed75dcac150d28cda03/packages/cli-kit/src/public/node/api/graphql.ts#L65).

I believe we could keep that in our new API, so I'd update this to something like this:

```suggestion
/**
 * Executes a GraphQL query against the Admin API. Uses typed documents.
 *
 * @param query - GraphQL query to execute.
 * @param session - Shopify admin session including token and Store FQDN.
 * @param variables - GraphQL variables to pass to the query.
 * @param version - API version.
 * @param responseOptions - Control how API responses will be handled.
 * @returns The response of the query of generic type <TResult>.
 */
export async function adminRequestDoc<TResult, TVariables extends Variables>(
  query: TypedDocumentNode<TResult, TVariables>,
  session: AdminSession,
  variables?: TVariables,
  version?: string,
  responseOptions?: GraphQLResponseOptions<TResult>,
): Promise<TResult> {
  let apiVersion = version
  if (!version) {
    apiVersion = await fetchLatestSupportedApiVersion(session)
  }
  const opts = {
    url: adminUrl(session.storeFqdn, apiVersion),
    api: 'Admin',
    token: session.token,
  }
  const result = graphqlRequestDoc<TResult, TVariables>({...opts, query, variables, responseOptions})
  return result
}
```

Reply:
Great catch, thank you!